warehouse/router/upload.go

```diff
@@ -427,6 +431,42 @@ func (job *UploadJob) run() (err error) {
 	return nil
 }
 
+func (job *UploadJob) cleanupObjectStorageFiles() error {
+	destination := job.warehouse.Destination
+	cleanupObjectStorageFiles := job.warehouse.GetBoolDestinationConfig(model.CleanupObjectStorageFilesSetting)
+	if !cleanupObjectStorageFiles {
+		return nil
+	}
+	storageProvider := whutils.ObjectStorageType(destination.DestinationDefinition.Name, destination.Config, job.upload.UseRudderStorage)
+	fm, err := filemanager.New(&filemanager.Settings{
+		Provider: storageProvider,
+		Config: misc.GetObjectStorageConfig(misc.ObjectStorageOptsT{
+			Provider:         storageProvider,
+			Config:           destination.Config,
+			UseRudderStorage: job.upload.UseRudderStorage,
+			WorkspaceID:      job.upload.WorkspaceID,
+		}),
+	})
+	if err != nil {
+		return fmt.Errorf("creating file manager: %w", err)
+	}
+	loadingFiles, err := job.loadFilesRepo.GetByStagingFiles(job.ctx, job.stagingFileIDs)
+	if err != nil {
+		return fmt.Errorf("fetching loading files: %w", err)
+	}
+	filesToDelete := make([]string, len(job.stagingFiles)+len(loadingFiles))
+	for i, file := range job.stagingFiles {
+		filesToDelete[i] = fm.GetDownloadKeyFromFileLocation(file.Location)
+	}
+	for i, file := range loadingFiles {
+		filesToDelete[i+len(job.stagingFiles)] = fm.GetDownloadKeyFromFileLocation(file.Location)
+	}
```

Comment:
minor: Can see if you like `lo` map here.

```
fileKeysToDel := lo.Map(job.stagingFiles, func(file FileType, _ int) string {
	return fm.GetDownloadKeyFromFileLocation(file.Location)
})
loadingFilesKeys := lo.Map(loadingFiles, func(file FileType, _ int) string {
	return fm.GetDownloadKeyFromFileLocation(file.Location)
})

fileKeysToDel = append(fileKeysToDel, loadingFilesKeys...)
```

Reply:
Replaced with `lo`